src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +192,55 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.push(
```

Comment:
Let's create a smarter, DRYer util function for that and make sure we handle edge cases. What happens when some values are undefined? eg. for thresholds?

Reply:
> Let's create a smarter, DRYer util function for that and make sure we handle edge cases. What happens when some values are undefined? eg. for thresholds?

I'll refactor the code once I get approval of my changes. 
Taking about the edge cases, I'm taking care of them already

```
    if (isHazardLayerId(hazardLayerParamId)) {
      setHazardLayerId(hazardLayerParamId);
    }
    if (isBaselineLayerId(baselineLayerParamId)) {
      setBaselineLayerId(baselineLayerParamId);
    }

    // Mean is selected by default
    if (statisticParam === 'median') {
      setStatistic(AggregationOperations.Median);
    }
```

Thresholds are optional, so I didn't add any checks for them. Also, they become undefined if they have an invalid number 
```
      threshold: {
        above: parseFloat(aboveThreshold) || undefined,
        below: parseFloat(belowThreshold) || undefined,
      },
```